main.js

```diff
@@ -178,4 +178,213 @@ const data = [
         hotel: 'Hotel Rehberge Berlin Mitte',
     },
 ];
-console.log(searchString('Mitte', data));
\ No newline at end of file
+console.log(searchString('M', data));
+
+function checkPalindrome(str) {
+return str.split('').reverse().join('') === str;
+}
+console.log(checkPalindrome('poloolop'))
+
+function searchHotels(str, hotels) {
+    let result = hotels.filter(function (e) {
```

Comment:
why are you using regular functions here? consider arrow functions, they are more commonly used as callbacks in such cases